src/Symfony/Component/Security/Guard/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -41,9 +41,14 @@
      * login page directly), this returns the URL the user should be redirected
      * to after logging in successfully (e.g. your homepage).
      *
+     * @deprecated Implement onAuthenticationSuccess() instead of needing this function.
+     *
      * @return string
      */
-    abstract protected function getDefaultSuccessRedirectUrl();
+    protected function getDefaultSuccessRedirectUrl()
```

Comment:
It will get executed if the user doesn't implement `onAuthenticationSuccess` *or* `getDefaultSuccessRedirectURL()`. In that case, this would be called here: 87. Since `onAuthenticationSuccess` is still implemented for BC, new users might not initially implement *either*, since there no interface/abstract method forces them to. It catches that case.

Also, deprecations are hard :).